src/bindings/python/src/openvino/frontend/pytorch/fx_decoder.py

```diff
@@ -453,3 +453,104 @@ def mark_node(self, node):
         node.set_friendly_name(name)
         super().mark_node(node)
         return node
+
+      
+class ExecuTorchPythonDecoder (TorchFXPythonDecoder):
+
+    # TODO: The constructor of ExecuTorchPythonDecoder is mostly similar to the
+    # constructor TorchFXTorchPythonDecoder. Update this to utilize a common
+    # implementation.
+    def __init__(self, pt_module, fx_gm=None, nodes=None,
+                 mark_node_callback=None, input_shapes=[], input_types=[], dynamic_shapes=False):
+        super().__init__(mark_node_callback)
+        self.pt_module = pt_module
+        self.fx_gm = fx_gm if fx_gm is not None else pt_module
+        self.input_types = [OVAny(pt_to_ov_type_map[str(t)])
+                            for t in input_types]
+        self.input_shapes = input_shapes
+
+        self._input_signature = []
+        self._example_input = None
+
+        if issubclass(type(pt_module), torch.fx.graph_module.GraphModule):
```

Comment:
```suggestion
        if isinstance(pt_module, torch.fx.graph_module.GraphModule):
```
can we do it?

Reply:
please create separate internal function in this class to handle `torch.fx.graph_module.GraphModule` case
and separate function to handle `torch.fx.Node` because now it is quite long for constructor

Reply:
I am working on both TorchFXPythonDecoder and ExecuTorchPythonDecoder constructor to address these issues and use helper functions to reduce redundant code. I will update the responses accordingly.

Reply:
I removed the constructor of ExecuTorchPythonDecoder and made these changes in TorchFXPythonDecoder which is also used by ExecuTorchPythonDecoder.